sympy/physics/vector/point.py

```diff
@@ -380,6 +380,7 @@ def v1pt_theory(self, otherpoint, outframe, interframe):
         v1 = self.vel(interframe)
         v2 = otherpoint.vel(outframe)
         omega = interframe.ang_vel_in(outframe)
+        otherpoint.set_vel(interframe, 0)
```

Comment:
Currently, the user has to ensure that `otherpoint` is fixed in `interframe` before calling `a2pt_theory()`. But users can pick the wrong point by accident, thinking it was fixed. I think it would be better to warn the user that `otherpoint` is not fixed in `interframe` at this location in the code. Overwriting this velocity will lead to confusion by users.

Reply:
Hi @moorepants, teaching this to my first year undergraduate students and this is definitely leading to a world of confusion for them that the `otherpoint` is not of zero velocity in the body that it is on. The 2 point theory is for two points on the same rigid body and therefore the `otherpoint` must have zero velocity in the `interframe`. So, I do not think that overwriting this velocity leads to confusion- if anything, I would argue that it leads to clarity that could be used for better bookkeeping of different points on a body.

Reply:
I just looked at this again to try to remember what it is about. I still have a similar concern. Right now, with most everything in the library, the user has to explicitly set the velocity of every point. The existing purpose of these velocity methods are really just helpers in calculating the velocity of a single point, it's never been expected that they establish velocities of multiple points. So, I'm still -1 to making this change without some examples of why it is necessary. There are lots of things that aren't set automatically for the user. We can of course start changing that, but we need to have good reason, provide examples of the necessity, and then document carefully any new behavior. Maybe start by providing an example problem where the current form of these methods causes issues. I personally haven't come across that yet, but maybe you are solving different problems or in a different way.

Reply:
Example is here: https://nbviewer.jupyter.org/github/angadhn/v1pt_changes/blob/main/Particle_on_Double_Pendulum.ipynb

TL; DR: velocities are not accessible for a points in all different frames.

Reply:
Thanks for the example. I see what you are expecting now. But you don't show why you need the velocities of those points set for any future steps in the problem. Knowing what necessitates these being set automatically could help me get on board.

The physics.vector design philosophy is effectively "every angular and linear velocity must be explicitly set by the user" and, as it currently stands, every command in sympy.physics.vector sets only a single velocity. There isn't the concept (yet) of an all-knowing system that is auto-magically making sure you do all your kinematics correctly. In fact, I think that is a feature, especially for learning. You want your students to control the specification of the positions, velocities, and accelerations and their relationships explicitly when learning. If we do more than that, we move towards a physics simulator that abstracts those details that are needed when teaching dynamics. For example, in the grad dynamics course I show them how to manually calculate v1/2pt relationships, then later tell them that there is this method that saves you a bit of typing.

For your example, if you need all those velocities set, this is the expected thing to type:

```python
O.set_vel(A, 0)
Q.set_vel(A, 0)
Q.v2pt_theory(O, N, A)
```

Every velocity is explicitly set by the user.  Each of those commands sets a single velocity.

I understand the desire to not have to type everything explicitly, but we've been trying to approach that by building on top of what's there (for example see the open pr's about adding a joint-type, which sets numerous velocities in one call).

The change, as proposed here, changes 10+ years of expected behavior from these methods and sets velocities that have never been set before. So I'm still not keen on merging this as proposed. One solution is to teach students how to make their own functions:

```python
def set_all_vel_with_v2pt(O, Q,  N, A):
    O.set_vel(A, 0)
    Q.set_vel(A, 0) 
    Q.set_vel(N, O.vel(N) + cross(A.ang_vel(N), Q.pos_from(O))
```
Then they write the function and know exactly what it does when they use it. This treats physics.vector as the base building block for the students to develop their proper kinematics methods.

Reply:
One more point is that these methods currently do exactly what they say in the docstring. The docstrings say anything about setting velocities on the other points passed in to the method. But maybe, there simply needs to be more clarity in the docstring of what the method does. Could that reduce confusion for your students?

Reply:
I chatted with @angadhn on the phone about this PR. I wanted to add somethings to look into:

- In the notebook he shared the last line is `O.vel(B)` which was never set explicitly yet it returns a value and we should check if that value is correct. It could be that this new change in #20049 causes that to be calculated.
- I don't think there is any precedence in the existing methods to set more than one velocity on any given method call on Point or ReferenceFrame. It has always been the case that calling `A.ang_vel_in(N)` will search the orientation connections between frames A and N and if there are rotation matrices fully connecting both, that the velocity will be calculated if it was never set. We very recently added that same feature, to be consistent, in PR #20049 for Points. But I don't think any intermediate velocities are set, only the single one that is requested. We should double check this though.